lib/msal-common/src/crypto/PopTokenGenerator.ts

```diff
@@ -49,16 +46,11 @@ export class PopTokenGenerator {
         };
     }
 
-    async signPopToken(accessToken: string, request: SignedHttpRequestParameters): Promise<string> {
-        const tokenClaims: TokenClaims | null = AuthToken.extractTokenClaims(accessToken, this.cryptoUtils);
```

Comment:
To handle a upgrade/downgrade scenario, we should consider still try this as a fallback if it is not found in the cache.

Reply:
Could you elaborate a bit? They `kid` is already being stored in the cache, so a missing `kid` would be an unexpected error even in an upgrade scenario.

Philosophically speaking, I think cracking open the access token was wrong to start with, but what are the advantages to using this as a fallback?

Reply:
Ah, nevermine, forgot it was already there.